### PR TITLE
Disable locality reodering in training

### PR DESCRIPTION
**Pull request by eellison**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #101423

cc @soumith @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @desertfire

Differential Revision: [D45874682](https://our.internmc.facebook.com/intern/diff/D45874682)

### CODE REVIEW

**Review by ngimel — APPROVED** [review]

**Comment by eellison** [comment]
@eellison has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D45874682).

File: torch/_inductor/compile_fx.py [hunk 1]
```diff
@@ -243,7 +243,10 @@ def compile_fx_inner(
     # correct we will need to fix.
 
     with V.set_fake_mode(fake_mode):
-        post_grad_passes(gm)
+        # has some issues with memory in training
+        locality_reorder = is_inference and config.reordering
```
Comment (jansel) [inline torch/_inductor/compile_fx.py:247]: Perhaps pass is_inference into `post_grad_passes`

Comment (eellison) [reply torch/_inductor/compile_fx.py:247]: will do in follow up since benchmarking isnt in a great state currently

**Comment by ezyang** [comment]
It is possible that locality reordering was responsible for a 7% perf improvement for resmlp_12_224, so would be good to turn this back on.

Evidence:

image

Inductor commits in this range

```
(/home/ezyang/local/a/pytorch-env) [ezyang@devgpu020.ftw1 ~/local/a/pytorch (effe1425)]$ git log --oneline 7dd8e08817ee59c926922409062e25f30408469b..6f7ebcdcd82afa7355e002c6e2b01f8da802db9b torch/_inductor
6f7ebcdcd82 [inductor] enable descriptive name for cpp kernels (#101330)
86869475ff9 [inductor] move dtype propagation log to schedule artifact (#101351)
dfc46153a72 [inductor] add graph id prefix to inductor_wrapper_call profile info (#101350)
7f3b00bfe07 [Inductor] Improve view/reshape on tensors with shape 0 (#101051)
1faef895ca4 (tag: ciflow/slow/1faef89) Inductor cpp wrapper: support sympy.Expr as input (#101257)
3b7c6b21d7a (tag: ciflow/slow/3b7c6b2) Disable locality reodering in training (#101423)
0a7ea9627f0 (tag: ciflow/periodic/100609, origin/gh/malfet/59/base) match sdpa patterns from HF (#100609)
a4830bd86b2 fix sign return type (#101346)
3e1c8168f8f (tag: ciflow/periodic/100713) Add pattern to merge/simplify split-cat (#100713)
87f9160b672 Revert "[inductor] fix incorrect strides in copy() decomp, fix hf_LongFormer + hf_BigBird errors (#100115)"
```

**Comment by eellison** [comment]
@ezyang I agree we should enable some form of locality reordering. we probably want to do it at the inductor scheduler level and it will need to be peak memory aware

**Pull request closed without merging**